### PR TITLE
readme for runing dataset_ingress_v1, plus SKIP_DATA_UPLOAD capability

### PR DESCRIPTION
**Pull request by mikeAdamss**
### What

- Added instructions for running `dataset_ingress_v1.py`.
- Added a new env var of `SKIP_DATA_UPLOAD` as we probaly dont want to upload files to s3 every time we run a transform we're developing.
- Fixed a mistake of mine where I said we dont need to raise errors to log them, we do, the logger needs the stack trace.

### How to review

Sanity check - **run the script** as per instructions and make sure it works for you.

### Who can review

Anyone.

### CODE REVIEW

File: dpypelines/pipeline/dataset_ingress_v1.py [hunk 1]
```diff
@@ -345,94 +349,127 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
...
+        elif skip_data_upload.strip() in ["False", "false"]:
+            skip_data_upload = False
+        else:
+            # Catch a trivial raise as we need the stack trace of the
+            # error for the logger, so it needs to be a raised error.
+            try:
+                raise ValueError(
+                    "If present SKIP_DATA_UPLOAD must be one of: True, true, False, false."
+                )
+            except ValueError as err:
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/dataset_ingress_v1.py]: Should this be `ValueError` or `Exception`?

File: dpypelines/pipeline/README.md [hunk 2]
```diff
@@ -4,33 +4,58 @@ Instructions on running existing pipelines can be found below. As more pipeline
...
+**Before** running this, you'll need to set some environment variables based on one of two scenarios as follows:
+
+### Scenario 1: You dont care about uploading the results of the pipeline (you just want to run the transform and see the outputs)
+
+This is probably the setup you'll want to use the majority of the time when developing locally.
+
+Set the following env vars:
+
+```
+export DISABLE_NOTIFICATIONS=true
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/README.md:44]: Need to set UPLOAD_SERVICE_URL as well (can be a nonsense string if not actually uploading)

File: dpypelines/pipeline/dataset_ingress_v1.py [hunk 3]
```diff
@@ -187,25 +187,29 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
...
-                        "submitter_email": submitter_email,
-                        "supplementary_distribution_pattern": supp_dist_pattern,
-                    },
-                )
-                de_notifier.failure()
-                raise err
+                # Catch a trivial raise as we need the stack trace of the
+                # error for the logger, so it needs to be a raised error.
+                try:
+                    raise FileNotFoundError(
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/dataset_ingress_v1.py:197]: Do we want to send an email at this point?

File: dpypelines/pipeline/dataset_ingress_v1.py [hunk 4]
```diff
@@ -345,94 +349,127 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
...
-    except Exception as err:
-        logger.error(
-            "Error creating UploadClient", err, data={"upload_url": upload_url}
-        )
-        de_notifier.failure()
-        raise err
+    # Allow DE's to skip the upload to s3 part of the pipeline while
+    # developing code locally.
+    skip_data_upload = os.environ.get("SKIP_DATA_UPLOAD", False)
+    if skip_data_upload is not False:
```
Comment (SarahJohnsonONS) [inline dpypelines/pipeline/dataset_ingress_v1.py:359]: Use `dpypelines.pipeline.shared.utils.str_to_bool` here?

File: dpypelines/pipeline/dataset_ingress_v1.py (hunk 3 above)
Comment (mikeAdamss) [reply dpypelines/pipeline/dataset_ingress_v1.py:197]: we sent if just below. By no means a hill I'll die on.

this is the bit I wrong about before - we need to have a raised error (so the logger can create info from the stack trace).

so we:
- raise it
- catch it
- log it
- send email
- raise it properly.

you could email just before here like you say, then raise and catch it to do the logger part, but seemed slightly neater having just the raise. By no means a hill I'd die on.

did that make sense or am I misunderstanding you?

File: dpypelines/pipeline/dataset_ingress_v1.py (hunk 4 above)
Comment (mikeAdamss) [reply dpypelines/pipeline/dataset_ingress_v1.py:359]: ty, I literally just added a card to add that function, forgot we made one 😆

File: dpypelines/pipeline/dataset_ingress_v1.py (hunk 1 above)
Comment (mikeAdamss) [reply dpypelines/pipeline/dataset_ingress_v1.py]: ValueError should be fine, its nice and explicit. Lets keep it handling just the scenario we're creating.

File: dpypelines/pipeline/dataset_ingress_v1.py (hunk 3 above)
Comment (SarahJohnsonONS) [reply dpypelines/pipeline/dataset_ingress_v1.py:197]: Makes sense, I've just never seen the `try: raise, except: raise again` structure before.

**Review by SarahJohnsonONS — APPROVED** [review]
LGTM

**Pull request merged**